### PR TITLE
[Doc] Add documentation for .replication-metadata-index for cross-cluster-replication.

### PR DESCRIPTION
**Pull request by skumarp7**
### Description
This PR adds `documentation for .replication-metadata-index for cross-cluster-replication.

### Issues Resolved
Fixes https://github.com/opensearch-project/cross-cluster-replication/issues/1033

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
@monusingh-1 - Hi Monu, can you please give this a technical review? Thanks!

**Review by monusingh-1 — APPROVED** [review]

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 1]
```diff
@@ -237,6 +237,13 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
   }]
 }
 ```
+### `.replication-metadata-store index`:
```
Comment (Naarcha-AWS) [inline _tuning-your-cluster/replication-plugin/getting-started.md:240]:
```suggestion
### `.replication-metadata-store` index
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 2]
```diff
@@ -237,6 +237,13 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
...
+
+1. `.replication-metadata-store` index serves as a persistent store for replication-related metadata and auto-follow rules within a cluster. It stores the replication metadata of each index getting replicated from the leader  to follower opensearch cluster.
```
Comment (Naarcha-AWS) [inline _tuning-your-cluster/replication-plugin/getting-started.md:242]:
```suggestion
The `.replication-metadata-store` index is a persistent data store for replication-related metadata and auto-follow rules inside of a cluster. It stores the replication metadata of each index getting replicated from the leading cluster to the follower cluster.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 3]
```diff
@@ -237,6 +237,13 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
...
+
+2. The index is created in the follower cluster upon the initiation of the first replication API trigger. Any updation/addition in the replication jobs/rules also get updated in the index. This enables the plugin to maintain a comprehensive record of replication status and rules across clusters.
```
Comment (Naarcha-AWS) [inline _tuning-your-cluster/replication-plugin/getting-started.md:244]:
```suggestion
After the first replication API trigger, the `.replication-metadata-store` index is created inside the follower cluster. Any update or addition to replication jobs or rules also get updated in the index. This enables the plugin to maintain a comprehensive record of replication status and rules across clusters.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 4]
```diff
@@ -237,6 +237,13 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
...
+   
+   Note: `.replication-metdata-store` is marked as a hidden index.
```
Comment (Naarcha-AWS) [inline _tuning-your-cluster/replication-plugin/getting-started.md:246]:
```suggestion
 `.replication-metdata-store` is a hidden index.
 {: .note}
```

Comment (Naarcha-AWS) [reply _tuning-your-cluster/replication-plugin/getting-started.md:246]: @skumarp7: Could you clarify the implications of this note? Do users need to unhide the index in order for it to be created during cluster replication?

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _tuning-your-cluster/replication-plugin/getting-started.md (hunk 4 above)
Comment (skumarp7) [reply _tuning-your-cluster/replication-plugin/getting-started.md:246]: Hi @Naarcha-AWS , There is no such implications, its just a hidden index. so users cannot view it when they use normal query to list the indices.

Comment (monusingh-1) [reply _tuning-your-cluster/replication-plugin/getting-started.md:246]: Yes, the index is a hidden system index which the users will not be able to query and is solely used by cross cluster replication plugin.

**Review by monusingh-1 — APPROVED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 5]
```diff
@@ -237,6 +237,14 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
   }]
 }
 ```
+### `.replication-metadata-store` index
+
+The `.replication-metadata-store` index is a persistent data store for replication-related metadata and auto-follow rules inside of a cluster. It stores the replication metadata of each index getting replicated from the leading cluster to the follower cluster.
```
Comment (natebower) [inline _tuning-your-cluster/replication-plugin/getting-started.md:242]:
```suggestion
The `.replication-metadata-store` index is a persistent data store for replication-related metadata and auto-follow rules inside of a cluster. It stores the replication metadata of each index being replicated from the leader cluster to the follower cluster.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 6]
```diff
@@ -237,6 +237,14 @@ curl -XGET -k -u 'admin:admin' 'https://localhost:9200/follower-01/_search?prett
...
+
+After the first replication API trigger, the `.replication-metadata-store` index is created inside the follower cluster. Any update or addition to replication jobs or rules also get updated in the index. This enables the plugin to maintain a comprehensive record of replication status and rules across clusters.
```
Comment (natebower) [inline _tuning-your-cluster/replication-plugin/getting-started.md:244]:
```suggestion
After the first replication API trigger, the `.replication-metadata-store` index is created inside the follower cluster. Any updates or additions to replication jobs or rules are also updated in the index. This enables the plugin to maintain a comprehensive record of replication status and rules across clusters.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Just a few changes. Thanks!

**Pull request merged**